src/dev/reg_bank.hh

```diff
@@ -955,6 +978,47 @@ class RegisterBank : public RegisterBankBase
         }
     }
 
+    template <class FillerReg>
+    void
+    addRegistersAt(std::initializer_list<RegisterAdder> adders)
+    {
+        panic_if(std::empty(adders),
+                "Adding an empty list of registers to %s?", name());
+
+        std::vector<RegisterAdder> vec{adders};
+        std::sort(vec.begin(), vec.end(),
+            [] (const auto& first, const auto& second) {
+                return first.offset.value() < second.offset.value();
+            }
+        );
+
+        for (auto &adder: vec) {
+            assert(adder.offset && adder.reg);
```

Comment:
~~Does this mean the offset cannot be 0?~~

~~I *think* that's what the code says, if this isn't intended you may want to take a closer look.~~

Ok, I see this is a `std::optional`. The code is OK, but I think it's confusing. It is not clear from this code (without spending the time to dig into the definition of `RegisterAdder` that calling `adder.offset` will give you whether or not an offset exists. Someone outside coming to this code would expect to find the `offset` value.

I suppose there's no action here, but I'll leave the note for other reviewers.

Reply:
Yeah I agree. To be fair, I didn't remove the old addRegisters flavour, but IMHO by adding this new utility, passing the { offset, reg } pair to addRegisters loses most of its meaning... If we removed it, there won't be any need to use optionals. The best way to make sure your offsets are respected is... to use addRegistersAt! Anyway I would defer this discussion until someone decides to deprecate it.